RFP-01.md

```diff
@@ -0,0 +1,175 @@
+
+### Key Features
+
+- Ultra gas efficient signed settlement function
+- No co-mingling of funds between solvers
+- Easily observable on-chain addresses for MEV Blocker rebates
+- Permissionless access to the batch auction with reduced bonding requirements
+
+### Deliverables
+
+- Contract system for `SubPoolFactory`, `SubPool`, and `SignedSettlement`
+- Comprehensive documentation
+- Open-source code repository
+
+### Interface
+
+```solidity
+interface Auth {
+    /**
+     * Allows `usr` to perform root-level functions
+     */
+    function addOwner(address usr) external auth;
+    /**
+     * Denies `usr` from being able to perform root-level functions
+     */
+    function removeOwner(address usr) external auth;
+    /**
+     * Determine if `usr` is able to perform root-level functions
+     */
+    function isOwner(address usr) external view returns (bool);
+}
+
+interface SubPool is Auth {
+    /**
+     * Determine the amount required to `heal` the pool
+     * @return amt of immutable collateralToken
+     * @return cowAmt of COW
+     */
+    function dues() external view returns (uint256 amt, uint256 cowAmt);
+    /**
+     * Pull in from `msg.sender` the dues required to `heal` the pool
+     */
+    function heal() external;
+}
+
+interface SubPoolFactory is Auth {
+    // --- view functions ---
+
+    /**
+     * View function for determining the minimum collateral required if using `token`
+     * @param token The token to retrieve the collateral requirement for
+     */
+    function collateral(IERC20 token) external view returns (uint256);
+    /**
+     * View function for determining if a given solver EOA has a correctly collateralized
+     * sub-pool and has not announced an exit.
+     * @param subPoolSolver The solver to check
+     * @return bool if the solver is allowed to solve
+     */
+    function canSolve(address subPoolSolver) external view returns (bool);
+    /**
+     * View function for determining the sub-pool for a given solver
+     * @param solver The solver to retrieve the sub-pool for
+     * @return SubPool
+     */
+    function solverSubPool(address solver) external view returns (SubPool);
+    /**
+     * View function for determining the backend URI for the sub-pool
+     * @return string URI
+     */
+    function backendUri(SubPool pool) external view returns (string memory);
+
+    // --- permissionless functions ---
+    /**
+     * Deploy a `SubPool` deterministically based on `msg.sender`.
+     * @notice The `msg.sender` SHOULD NOT be a contract.
```

Comment:
This specification is written for someone executing on a grant. "SHOULD" means they have leeway (this isn't a protocol specification - it's a grant - this isn't an interface of a multilateral agreement among many, many stakeholders).

So, **if** the implementer wants to include contracts, so be it, but it would be on them to ensure that vulnerabilities are assessed / taken care of.

Reply:
To me this is a protocol-level decision: a grantee has no way to judge if in the full scope of the protocol having a smart-contract solver could be an issue. Also, most concerns on the protocol side are imho for ancillary tools like analytics that might be assuming something about the transaction sender, or that there can be at most a settlement per transaction.
Pragmatically, a grantee isn't going to take the time to weight these considerations. So it's on us: do we want it or not?

Reply:
I don't think this is a _protocol-level_ decision but is a _bonding pool_ level decision. By virtue of having the `SignedSettlement` contract, running under CoW DAO's bonding pool, the protocol-level decision is implicitly made.

Having said this, I know that the multiple settlements per transaction have historically caused issues and this is an easy one to overlook, so I'd be inclined towards making this a **MUST NOT** be a contract?